12875530: add documentation for CoF (#10)
rabobank-nederland/omnikassa-sdk-doc: README.md

| `initiatingParty`               | An ID identifying the party from which the order announcement was initiated                                           | Optional. This field must be left empty unless agreed otherwise with Rabobank                                                                                                                                                                                                                    |
| `skipHppResultPage`             | Use this field to skip the hosted result page (also referred to as the success/thank you page) in the payment process | Optional                                                                                                                                                                                                                                                                                         |

> | `shopperRef`                    | This is the reference of the customer at merchant webshop, should be a uniq value at merchant webshop                 | Optional                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| `shopperRef`                    | This is the reference of the customer at merchant webshop, should be a unique value at merchant webshop                 | Optional                                                                                                                                                                                                                                                                                         |
```

the webshop. The success page (also referred to as the “Thank You” page) of the hosted payment pages will be skipped in
this case.

<a name="cards-on-file"></a>
#### Allow customers to save credit cards at rabo smart pay

> Smart pay now enables customers to save their cards for a webshop, and this can speed up the checkout process when

Reviewer: ```suggestion
Smart pay enables customers to save their cards for a webshop, and this can speed up the checkout process when
```

I was thinking that this might make it more future proof

customers come back to the same webshop using the same same credit card for checkout.

This feature is enabled if:

> 1. Webshop is onboarded with Cards on File feature;

Reviewer: ```suggestion
1. Webshop is configured via the merchant dashboard for the Cards on File feature;
```

2. During order announcement, a `shopperRef` is provided and `emailAddress` of `customerInformation` is provided;

> 3. The `paymentBrandMetaData` map has `enableCardOnFile` as `true`;

Reviewer: ```suggestion
3. The value of `paymentBrandMetaData.enableCardOnFile` is `true`;
```

Merchants will be able to manage the saved credit cards on behave of customers if necessary
**Java**
```java

> //Gate list of cards saved by customer

Reviewer: ```suggestion
//Get a list of cards saved by customer
```

ShopperPaymentDetailsResponse paymentDetailsResponse = endpoint.getShopperPaymentDetails(shopperRef);
//Delete a saved card for a customer, id is the id of cardOnFile
endpoint.deleteShopperPaymentDetails(shopperRef, id);
```

**ShopperPaymentDetailsResponse**

| Fields          | Description                                   |
|-----------------|-----------------------------------------------|

> | cardOnFileList  | List of `cardOnFile` saved by rabo smart pay  |

Reviewer: ```suggestion
| cardOnFileList  | List of `cardOnFile` entries saved by rabo smart pay  |
```

**cardOnFile**

| Fields      | Decription                                                                                           |
|-------------|------------------------------------------------------------------------------------------------------|
| id          | id of the card saved by smart pay, must be provided when the merchant wants to delete the saved card |
| last4Digits | last 4 digits of the card                                                                            |
| brand       | card brand, e.g. VISA                                                                                |

> | cardExpiry  | the expiry of the card. Will be format YYYY-MM                                                       |

Reviewer: ```suggestion
| cardExpiry  | the expiry year and month of the card. Will be format YYYY-MM                                                       |
```

> | tokenExpiry | the expiry of the token. Will be format YYYY-MM                                                      |

Reviewer: ```suggestion
| tokenExpiry | the expiry year and month of the token. Will be format YYYY-MM                                                      |
```
